src/Commands/CloneCommand.php

```diff
@@ -0,0 +1,217 @@
+    {
+        return $site->environments
+            ->get($this->targetEnv)
+            ->$opMethod($this->originEnv);
+    }
+
+    /**
+     * Setup the progress bar.
+     *
+     * @param string $operation
+     *
+     * @return \Symfony\Component\Console\Helper\ProgressBar
+     */
+    protected function startProgressBar($operation = '')
+    {
+        ProgressBar::setFormatDefinition('minimal', self::PROGRESS_FORMAT);
+        $progress = $this->io()->createProgressBar(100);
+        $progress->setFormat('minimal');
+        $progress->setMessage("Cloning {$operation}...");
+        $progress->start();
+        return $progress;
+    }
+
+    /**
+     * Refresh Workflow data, advance progress, and hangout.
+     *
+     * @param \Terminus\Models\Workflow $workflow
+     * @param \Symfony\Component\Console\Helper\ProgressBar $progress
+     *
+     * @return bool
+     */
+    protected function doWorkflow(Workflow $workflow, ProgressBar $progress)
+    {
+        $workflow->fetch();
+        $progress->advance(1);
+        sleep(1);
+        return true;
+    }
+
+    private function setOperations(array $options = [])
+    {
+        $this->operations = $options;
+        return $this;
+    }
+
+    /**
+     * @return array Operations to be performed.
+     */
+    private function getOperations()
+    {
+        return $this->operations;
+    }
+
+    /**
+     * Check to see that origin and target are from the same site.
+     *
+     * @param string $origin
+     * @param string $target
+     *
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateSites($origin, $target)
+    {
+        if (explode('.', $origin)[0] != explode('.', $target)[0]) {
+            throw new \Exception("Clone operations are only available within the same site.", 1);
+        }
+        return true;
+    }
+
+    /**
+     * Set the origin environment.
+     *
+     * @param string $env
+     *
+     * @return $this
+     */
+    private function setOriginEnv($env = '')
+    {
+        $this->originEnv = $env;
+        return $this;
+    }
+
+    /**
+     * Set the target environment.
+     *
+     * @param string $env
+     *
+     * @return $this
+     */
+    private function setTargetEnv($env = '')
+    {
+        $this->targetEnv = $env;
+        return $this;
+    }
+
+    /**
+     * Set the site name.
+     *
+     * @param string $name
+     *
+     * @return $this
+     */
+    private function setSiteName($name = '')
+    {
+        $this->siteName = $name;
+        return $this;
+    }
+
+    /**
+     * Fetch the method name for a workflow operation based on a short name.
+     *
+     * @param string $operation
+     *
+     * @return string
+     * @throws \Exception
+     */
+    private function formatOpMethod($operation = '')
+    {
+        if (!array_key_exists($operation, $this->methods)) {
+            throw new \Exception("{$operation} does not have an available clone method.", 1);
```

Comment:
We are not validating inputs in this phase. Using this method dictionary makes the processes being run less clear.